### PR TITLE
UIDEXP-196: Rewrite Bigtest tests for mapping profile details with jest/react-testing-library

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
## Purpose
https://issues.folio.org/browse/UIDEXP-196

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by vashjs — APPROVED** [review]

File: src/settings/MappingProfiles/MappingProfileDetails/MappingProfileDetails.test.js [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+          allTransformations={allMappingProfilesTransformations}
+          isDefaultProfile
+          isProfileUsed
+        />,
+        translationsProperties
+      );
+      expect(screen.getByRole('button', { name: /collapse all/i })).toBeVisible();
+    });
+    it('should display accordions set', () => {
+      const { container } = renderWithIntl(
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/settings/MappingProfiles/MappingProfileDetails/MappingProfileDetails.test.js:108]: Yes, you are right. I will fix right now!

**Review by axelhunn — APPROVED** [review]

File: src/settings/MappingProfiles/MappingProfileDetails/MappingProfileDetails.test.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+      const summaryNameValue = container.querySelector('[data-test-mapping-profile-name]');
+      const summaryDescriptionValue = container.querySelector('[data-test-mapping-profile-description]');
+      const summaryOutputFormat = container.querySelector('[data-test-mapping-profile-output-format]');
+      const summaryRecordType = container.querySelector('[data-test-mapping-profile-folio-record-type]');
+
+      expect(getByText(summaryNameValue, 'AP Holdings 1')).toBeVisible();
+      expect(getByText(summaryDescriptionValue, 'AP Holdings 1 description')).toBeVisible();
+      expect(getByText(summaryOutputFormat, 'MARC')).toBeVisible();
+      expect(getByText(summaryRecordType, commonTranslations['recordTypes.holdings'])).toBeVisible();
+    });
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/settings/MappingProfiles/MappingProfileDetails/MappingProfileDetails.test.js]: Yes, I agree with you. You showed me great options for making Jest/RLT tests better in future. I already fixed that test cases according your comment. I really appreciate to  you for that.

File: src/settings/MappingProfiles/MappingProfileDetails/MappingProfileDetails.test.js [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+        expect(deleteConfirmationModal).toBeVisible();
+      });
+
+      it('should hide delete confirmation modal', () => {
+        setupActionDeleteClick();
+        userEvent.click(screen.getByText(/cancel/i));
+
+        return waitForElementToBeRemoved(() => document.querySelector('#delete-mapping-profile-confirmation-modal'));
+      });
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/settings/MappingProfiles/MappingProfileDetails/MappingProfileDetails.test.js]: Thanks a lot! Already fixed it.

File: src/settings/MappingProfiles/MappingProfileDetails/MappingProfileDetails.test.js [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+        stripes={stripes}
+        mappingProfile={mappingProfile}
+        isDefaultProfile={isDefaultProfile}
+        isProfileUsed={isProfileUsed}
+        onCancel={onCancel}
+        onEdit={onEdit}
+        onDelete={onDelete}
+      />
+    </SettingsComponentBuilder>
+
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/settings/MappingProfiles/MappingProfileDetails/MappingProfileDetails.test.js]: Thanks, fixed it.

**Pull request merged**